meson.build

```diff
@@ -0,0 +1,439 @@
+  'Source/C++/Core/Ap4MehdAtom.h',
+  'Source/C++/Core/Ap4MfhdAtom.h',
+  'Source/C++/Core/Ap4MfroAtom.h',
+  'Source/C++/Core/Ap4MoovAtom.h',
+  'Source/C++/Core/Ap4Movie.h',
+  'Source/C++/Core/Ap4MovieFragment.h',
+  'Source/C++/Core/Ap4Mpeg2Ts.h',
+  'Source/C++/Core/Ap4MvhdAtom.h',
+  'Source/C++/Core/Ap4NmhdAtom.h',
+  'Source/C++/Core/Ap4ObjectDescriptor.h',
+  'Source/C++/Core/Ap4OdafAtom.h',
+  'Source/C++/Core/Ap4OddaAtom.h',
+  'Source/C++/Core/Ap4OdheAtom.h',
+  'Source/C++/Core/Ap4OhdrAtom.h',
+  'Source/C++/Core/Ap4OmaDcf.h',
+  'Source/C++/Core/Ap4PdinAtom.h',
+  'Source/C++/Core/Ap4Piff.h',
+  'Source/C++/Core/Ap4Processor.h',
+  'Source/C++/Core/Ap4Protection.h',
+  'Source/C++/Core/Ap4PsshAtom.h',
+  'Source/C++/Core/Ap4Results.h',
+  'Source/C++/Core/Ap4RtpAtom.h',
+  'Source/C++/Core/Ap4RtpHint.h',
+  'Source/C++/Core/Ap4SaioAtom.h',
+  'Source/C++/Core/Ap4SaizAtom.h',
+  'Source/C++/Core/Ap4Sample.h',
+  'Source/C++/Core/Ap4SampleDescription.h',
+  'Source/C++/Core/Ap4SampleEntry.h',
+  'Source/C++/Core/Ap4SampleSource.h',
+  'Source/C++/Core/Ap4SampleTable.h',
+  'Source/C++/Core/Ap4SbgpAtom.h',
+  'Source/C++/Core/Ap4SchmAtom.h',
+  'Source/C++/Core/Ap4SdpAtom.h',
+  'Source/C++/Core/Ap4SegmentBuilder.h',
+  'Source/C++/Core/Ap4SencAtom.h',
+  'Source/C++/Core/Ap4SgpdAtom.h',
+  'Source/C++/Core/Ap4SidxAtom.h',
+  'Source/C++/Core/Ap4SLConfigDescriptor.h',
+  'Source/C++/Core/Ap4SmhdAtom.h',
+  'Source/C++/Core/Ap4StcoAtom.h',
+  'Source/C++/Core/Ap4SthdAtom.h',
+  'Source/C++/Core/Ap4String.h',
+  'Source/C++/Core/Ap4StscAtom.h',
+  'Source/C++/Core/Ap4StsdAtom.h',
+  'Source/C++/Core/Ap4StssAtom.h',
+  'Source/C++/Core/Ap4StszAtom.h',
+  'Source/C++/Core/Ap4SttsAtom.h',
+  'Source/C++/Core/Ap4Stz2Atom.h',
+  'Source/C++/Core/Ap4SyntheticSampleTable.h',
+  'Source/C++/Core/Ap4TencAtom.h',
+  'Source/C++/Core/Ap4TfdtAtom.h',
+  'Source/C++/Core/Ap4TfhdAtom.h',
+  'Source/C++/Core/Ap4TfraAtom.h',
+  'Source/C++/Core/Ap4TimsAtom.h',
+  'Source/C++/Core/Ap4TkhdAtom.h',
+  'Source/C++/Core/Ap4Track.h',
+  'Source/C++/Core/Ap4TrakAtom.h',
+  'Source/C++/Core/Ap4TrefTypeAtom.h',
+  'Source/C++/Core/Ap4TrexAtom.h',
+  'Source/C++/Core/Ap4TrunAtom.h',
+  'Source/C++/Core/Ap4Types.h',
+  'Source/C++/Core/Ap4UrlAtom.h',
+  'Source/C++/Core/Ap4Utils.h',
+  'Source/C++/Core/Ap4UuidAtom.h',
+  'Source/C++/Core/Ap4Version.h',
+  'Source/C++/Core/Ap4VmhdAtom.h',
+  'Source/C++/Core/Ap4VpccAtom.h',
+  'Source/C++/Crypto/Ap4AesBlockCipher.h',
+  'Source/C++/Crypto/Ap4Hmac.h',
+  'Source/C++/Crypto/Ap4KeyWrap.h',
+  'Source/C++/Crypto/Ap4StreamCipher.h',
+  'Source/C++/MetaData/Ap4MetaData.h'
+]
+
+ap4_include = [
+  include_directories('Source/C++/Codecs'),
+  include_directories('Source/C++/Core'),
+  include_directories('Source/C++/Crypto'),
+  include_directories('Source/C++/MetaData')
+]
+
+if get_option('default_library') == 'both'
+  error('default library cannot be set to both, it can be either static or shared')
+endif
```

Comment:
This should only be true on Windows / with MSVC, probably?

Reply:
I have tested to build with default library set to both and it works. I wasn't sure meson will link executables to ap4 shared library by default. Now I have fixed it